src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,228 @@
 package core.basesyntax;
 
+import java.util.Map;
+import java.util.Objects;
+
+import static core.basesyntax.MyHashMap.Node.hash;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static int CAPACITY = 16;
+    private Node<K, V>[] table;
+    private int size;
+    private int threshold;
+
+    public MyHashMap() {
+        table = new Node[CAPACITY];
+        threshold = (int) (CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size > threshold) {
+            // resize();
+            //return;
+        }
+        int index = hash(key);
+        Node<K, V> newNode = new Node<>(key, value, null);
+        Node<K, V> node = table[index];
+        //linked list is empty
+        if (node == null) {
+            table[index] = newNode;
+            size++;
+            //linked list isn't empty
+        } else {
+            //оно не заходит сюда
+            //node.key.equals(key)
+            while (node.next != null || checkKey(key, node)) {
```

Comment:
The condition in the while loop seems incorrect. You are using the logical OR operator `||` which will cause the loop to continue if either condition is true. You should use the logical AND operator `&&` instead to ensure that you check the key only when there is a next node.